### ISSUE TITLE
Soundkiller doesn't unmute applications

### ISSUE THREAD

**Issue by JonathanHolvey** [jugglingcats/soundkiller#1]
Hi,

I've been using Soundkiller for a few weeks (after previously uninstalling because of the same issue) and I'm having a problem where the applications aren't being unmuted after returning to a user's desktop, like so:

![image](https://cloud.githubusercontent.com/assets/10114731/6245223/02274f46-b7a8-11e4-8736-c45983372f16.png)

Have you seen this problem before? Is there a way I can generate a log to help you debug it? I think this is a great little app, but it's a pain (and unintuitive) to have to unmute all the applications after switching users.

Edit: I'm using Windows 7 Pro 64-bit.

Cheers,
Jon

**Comment by jugglingcats** [comment]
Hi Jon, thanks for the feedback. Are you finding this happens all the time when switching users or just occasionally? I've definitely seen it happen but we run SoundKiller on our home machine and the kids are switching constantly and without any issue most of the time.

I'll try and find some time to add a debug option but can't promise anything as I have a busy day job!

**Comment by JonathanHolvey** [comment]
Hi,

Thanks for the quick reply.

I'm pretty sure it's happening all the time. I've just spent about 20 minutes switching users and observing what happens. The only time it behaved differently was immediately after a reboot, when after switching users and switching back, only the System Sounds entry in the volume mixer was muted. All subsequent times I tried (after manually unmuting) all three entries remained muted, as per the screenshot. It's also not limited to these applications.

I was having a quick look through your code as well, but I'm really only a casual hobby programmer and not familliar with C#. It may take a little while for me to get my head round it.

Cheers,
Jon

**Comment by jugglingcats** [comment]
The code to handle the user switch is actually super simple... it's here: https://github.com/jugglingcats/soundkiller/blob/master/AppMain/Program.cs.

It tries to mute all active apps (those using sound) on switch away and then unmute them on switch back (regardless if they were muted before).

There's only a couple of things that can go wrong:
- App not being reported as "active" in which case it will be ignored
- Exception being thrown when trying to mute/unmute (which is just ignored without notification)

It's odd that it's managing to mute but not unmute again...

The code could be smarter. It could record the sessions it muted and explicitly unmute the same sessions again on switch. This would mean for example that if an app was muted already on switching it wouldn't be unmuted on switching back. But it would make the code a fair bit more complex.

I assume all users are running the app in the background (I think you'd have a different issue if not anyway).

Are you Windows 7 or Windows 8?

**Comment by JonathanHolvey** [comment]
Yeah, that's the file I was looking through. The structure seems pretty straightforward, but it's difficult to tell exactly what the methods you're using are doing without looking up the documentation.

I'm on Windows 7 Pro 64-bit.
